### ISSUE TITLE
Tracker Failed:  IL Corsaro Nero

### ISSUE THREAD

**Issue by neuman1812** [Jackett/Jackett#1431]
05-30 09:47:55 Error got NotImplemented/InternalServerError
May 30 09:47:55 raspberry mono[15040]: 05-30 09:47:55 Error request.Method: GET
May 30 09:47:55 raspberry mono[15040]: 05-30 09:47:55 Error request.Url: https://ilcorsaronero.info/argh.php?search=Jeremy%20Clarkson%20Speed%20S01
May 30 09:47:55 raspberry mono[15040]: 05-30 09:47:55 Error request.Cookies:
May 30 09:47:55 raspberry mono[15040]: 05-30 09:47:55 Error request.Referer:
May 30 09:47:55 raspberry mono[15040]: 05-30 09:47:55 Error request.RawPOSTDdata:
May 30 09:47:55 raspberry mono[15040]: 05-30 09:47:55 Error cookies:
May 30 09:47:55 raspberry mono[15040]: 05-30 09:47:55 Error headerString:
May 30 09:47:55 raspberry mono[15040]:  
May 30 09:47:55 raspberry mono[15040]: 05-30 09:47:55 Info Found 0 releases from Il Corsaro Nero
-------------------------------

Jackett Version 0.7.1437.0

**Comment by kaso17** [comment]
Please try updating to mono 5 and post the output of `curl -v https://ilcorsaronero.info`

**Comment by kaso17** [comment]
never mind, please try updating to jackett v0.7.1469

**Comment by SBado** [comment]
Updated Jackett to v0.7.1469, still same error

**Comment by kaso17** [comment]
also make sure that your using mono 5

**Comment by neuman1812** [comment]
I triggered the update client,  but now Jacket it stuck in a loop loading cardigann definitions.    It goes through this loop after pausing at the code below for about 10 seconds.

```
Loading Cardigann definition /opt/jackett/Definitions/ztracker.yml 
2017-06-04 06:59:20.3177 Info Adding aggregate indexer 
2017-06-04 06:59:30.2177 Info App config/log directory: /home/pi/.config/Jackett 
2017-06-04 06:59:30.8707 Info Starting Jackett 0.7.1469.0 
2017-06-04 06:59:30.8740 Info Environment version: 4.0.30319.42000 (/usr/lib/mono/4.5) 
2017-06-04 06:59:30.8756 Info OS version: Unix 4.4.38.7 
2017-06-04 06:59:30.8773 Info ThreadPool MaxThreads: 400 workerThreads, 200 completionPortThreads 
2017-06-04 06:59:30.8786 Info issue: Ubuntu 16.04.2 LTS \n \l 
2017-06-04 06:59:30.8798 Info mono version: 5.0.1.1 (2017-02/5077205 Thu May 25 10:07:22 UTC 2017) 
2017-06-04 06:59:30.9152 Info Using HTTP Client: UnixLibCurlWebClient 
2017-06-04 06:59:35.7272 Info Loading Cardigann definitions from: /home/pi/.config/cardigann/definitions/ 
```

**Comment by SBado** [comment]
Still same error, even with Mono 5 (5.0.1.1)

**Comment by kaso17** [comment]
@neuman1812 seem like it's crashing and restarting, please try running it manually from the console
@SBado try forcing httpclient using the `-c httpclient` argument

**Comment by SBado** [comment]
If I force httpclient, I get this error:

An error occurred while sending the request: Error: TrustFailure (Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED at /home/abuild/rpmbuild/BUILD/mono-5.0.1.1/external/boringssl/ssl/handshake_client.c:1132)

This is the reason why I was using  libcurl. Before you ask me, the package ca-certificates is installed.

**Comment by kaso17** [comment]
@SBado Try running `wget -O - https://curl.haxx.se/ca/cacert.pem | cert-sync /dev/stdin` as root to import the CA certs to mono.

**Comment by SBado** [comment]
Already tried twice... :P

**Comment by neuman1812** [comment]
--Im going to open a new ticket.  I've run into some other issues now with failing.

**Comment by kaso17** [comment]
@SBado can you post the output of the cert-sync tool please?
running `wget -O - https://curl.haxx.se/ca/cacert.pem | cert-sync --user /dev/stdin` (with --user argument) as your jackett use might be an alternative too.

**Comment by SBado** [comment]
Output of cert-sync:

```
Mono Certificate Store Sync - version 5.0.1.1
Populate Mono certificate store from a concatenated list of certificates.
Copyright 2002, 2003 Motus Technologies. Copyright 2004-2008 Novell. BSD license                                                                                        d.

--2017-06-05 15:35:29--  https://curl.haxx.se/ca/cacert.pem
Risoluzione di curl.haxx.se (curl.haxx.se)... 151.101.38.49, 2a04:4e42:9::561
Connessione a curl.haxx.se (curl.haxx.se)|151.101.38.49|:443... connesso.
Richiesta HTTP inviata, in attesa di risposta... 200 OK
Lunghezza: 261889 (256K) [application/x-pem-file]
Salvataggio in: "STDOUT"

-                   100%[===================>] 255,75K  --.-KB/s    in 0,1s

2017-06-05 15:35:29 (1,83 MB/s) - scritto su stdout [261889/261889]

Importing into legacy system store:
I already trust 158, your new list has 158
Import process completed.

Importing into BTLS system store:
I already trust 157, your new list has 158
Certificate added: C=IL, O=StartCom Ltd., OU=Secure Digital Certificate Signing,                                                                                         CN=StartCom Certification Authority
1 new root certificates were added to your trust store.
Import process completed.
```

Tried also with `--user`, same result.

**Comment by kaso17** [comment]
And after restarting jackett you're still getting a "Error: TrustFailure" error?

**Comment by SBado** [comment]
Yes, correct

**Comment by kaso17** [comment]
Please try running jackett with disabled certificate validation (`--IgnoreSslErrors true` argument)

**Comment by SBado** [comment]
I tried:
```
06-05 16:25:42 Info Adding aggregate indexer
06-05 16:25:42 Info HttpWebClient: Disabling certificate validation
06-05 16:25:42 Info Starting web server at http://*:9117/
```
but still, the error keeps popping up. You can find the full stack trace [here](https://pastebin.com/mXNp42ZD), if you need it.

**Comment by SBado** [comment]
I deleted ~/.config/Jackett and I started everything from scratch and now it's working! Thanks for your help!

**Comment by SBado** [comment]
Do you need me to try something else? I also tried all the possible MONO_TLS_PROVIDER values with no success.

**Comment by flightlevel** [comment]
What happens if you run 
```
csharp -e 'Console.WriteLine (new System.Net.WebClient ().DownloadString ("https://www.howsmyssl.com/").IndexOf ("1.2"))'
```

**Comment by SBado** [comment]
```
csharp -e 'Console.WriteLine (new System.Net.WebClient ().DownloadString ("https://www.howsmyssl.com/").IndexOf ("1.2"))'
2393
```

**Comment by kaso17** [comment]
Any chance that your ISP or some fancy device is messing with HTTPS connections?
Please post the output of `curl -v https://ilcorsaronero.info`

**Comment by SBado** [comment]
Maybe this? ` SSL certificate problem: unable to get local issuer certificate`

```
curl -v https://ilcorsaronero.info
* Rebuilt URL to: https://ilcorsaronero.info/
*   Trying 91.213.8.120...
* TCP_NODELAY set
* Connected to ilcorsaronero.info (91.213.8.120) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (OUT), TLS alert, Server hello (2):
* SSL certificate problem: unable to get local issuer certificate
* stopped the pause stream!
* Closing connection 0
* TLSv1.2 (OUT), TLS alert, Client hello (1):
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
HTTPS-proxy has similar options --proxy-cacert and --proxy-insecure.
```

**Comment by kaso17** [comment]
Ok, they don't send the intermediate CA certificate. See https://www.ssllabs.com/ssltest/analyze.html?d=ilcorsaronero.info too.
You might want to inform the staff about this so they can fix it.
In the meantime 9870d38cbb7ff4b0806a089c88e9d508a4500d2b should workaround the problem.

**Comment by SBado** [comment]
Thank you very much! I'll try to get in touch with them and inform them. Thank you again!

**Comment by kaso17** [comment]
fixed again via f299cf3ac363cb51e13feefb349c96f3aafdbf97
This will break every few weeks until they fix their certificates, please point the staff at the incomplete CA chain issue: https://www.ssllabs.com/ssltest/analyze.html?d=ilcorsaronero.info&latest

**Comment by SBado** [comment]
I already informed them back in June and apparently they solved their issues. Yesterday, I noticed the tracker wasn't working anymore, so I went straight to ssllabs and I saw the CA chain was incomplete, again.
I wrote another email but they didn't reply yet. In case somebody else wants to inform them, this is their email address: italianbits@gmail.com.
Thank you.
